autosklearn/data/feature_validator.py

```diff
@@ -363,7 +357,7 @@ def list_to_dataframe(
             # Warn the user about dtypes or request him to use a dataframe
             for col in X_train.columns:
                 if X_train[col].dtype.name == 'string':
-                    X_train[col] = X_train[col].astype('category')
+                    X_train[col] = X_train[col].astype('string')
```

Comment:
Do we actually have to change something here or can we just drop the whole for-loop?

Reply:
This for-loop does seem unnecessary, if `dtype.name == 'string'` then there's no need to convert it.